src/main/resources/GoogleApps/Groovy.groovy

```diff
@@ -365,27 +368,32 @@ public class GoogleAppsGroovy {
             }
         }
         String id = user.get("resourceName");
-        if(id.startsWith("people/") && id.length()>7) id = id.substring(7);
+        if(id.startsWith("people/") && id.length()>7) {
+            id = id.substring(7);
+        }
         def db = context.getDatabase()
         try {
             // Force main wiki database to create the user as global
             context.setDatabase("xwiki")
-            if(email == "")
-                email = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            if(usersEmailAddress == "") {
+                usersEmailAddress = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            }
             def wikiUserList = services.query.xwql("from doc.object(GoogleApps.GoogleAppsAuthClass) as auth where auth.id=:id").bindValue("id", id).execute()
             if ((wikiUserList==null) || (wikiUserList.size()==0))
```

Comment:
I'm not sure you will get ever a null from https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-query/xwiki-platform-query-xwql/src/main/java/org/xwiki/query/xwql/internal/XWQLQueryExecutor.java#L79.
It will always be a list, maybe an empty one, but never a null one. Checking for nulls everywhere is unnecessary loading the code. WDYT?

Reply:
While I'd agree it's not nice to read, I don't read in the source of XWQLQueryExecutor that this is for sure not null (it returns the result of another variable).
Same remark as above: we should not try to change the groovy code too much now and focus on such enhancements for the java part.